src/Tasks/GenerateResource.cs

```diff
@@ -725,7 +725,15 @@ public override bool Execute()
                 }
                 else if (FailIfNotIncremental)
                 {
-                    Log.LogErrorFromResources("GenerateResource.OutOfDate");
+                    int maxCount = Math.Min(inputsToProcess.Count, outputsToProcess.Count);
+                    maxCount = Math.Min(maxCount, 5);  // Limit to just 5
+
+                    for (int index = 0; index < maxCount; index++)
+                    {
+                        Log.LogErrorFromResources("GenerateResource.ProcessingFile", inputsToProcess[index], outputsToProcess[index]);
```

Comment:
https://github.com/dotnet/msbuild/pull/8881#discussion_r1238428633

I'm still opinionated that adding a new resource string that explicitly mentions the problem (failing the incrementaility check) is easy and should have no downsides, while reusing the message that doesn't explain the problem (`"Processing resource file "{0}" into "{1}"."`) can easily lead to confusion - among users, but as well among the maintainers of MSBuild investigatin customers issues.

Reply:
I am in the camp where this feature is like "Treat Warning As Error" where it converts warning messages into errors, it tends to avoid generating a new message in most cases.  
Likewise, this feature converts non-skipped task/targets messages into errors.  I will add extra messages if none exists, but most tasks already has one.

Reply:
Maybe to help with user confusion is to print a general message at the end of build.  Ex:
`MSB9999: Incremental checker has found issues, see the errors in the log.`

Reply:
General error message at the end that clarifies the situation works for me as well

Reply:
Sorry, had to context switch due to deadlines.  I will see if I can squeeze it in before the Thanksgiving.

Reply:
Added.
`Question build Failed.  The build exited early upon detection of an incremental issue.`